pkg/mutation/assignmeta_mutator.go

```diff
@@ -78,10 +81,18 @@ func (m *AssignMetadataMutator) DeepCopy() Mutator {
 	res := &AssignMetadataMutator{
 		id:             m.id,
 		assignMetadata: m.assignMetadata.DeepCopy(),
+		path: &parser.Path{
+			Nodes: make([]parser.Node, len(m.path.Nodes)),
+		},
 	}
+	copy(res.path.Nodes, m.path.Nodes)
```

Comment:
It occurs to me that the Nodes themselves may need deepcopy methods, as they have some pointers.

Filed #993 to track